src/models/TalkMapper.php

```diff
@@ -589,6 +591,56 @@ public function thisUserHasAdminOn($talk_id)
         return false;
     }
 
+    /**
+     * Add talk to track
+     *
+     * @param int $talk_id
+     * @param int $track_id
+     *
+     * @return int
+     */
+    public function addTalkToTrack($talk_id, $track_id)
+    {
+        $params = [
+            'track_id' => $track_id,
+            'talk_id' => $talk_id,
+        ];
+
+        // is this link in the database already?
+        $sql = 'select ID from talk_track where track_id = :track_id and talk_id = :talk_id';
+        $stmt = $this->_db->prepare($sql);
+        $stmt->execute($params);
+        $talk_track_id = $stmt->fetchColumn();
+        
+        if ($talk_track_id === false) {
+            // insert new row as not in database
+            $sql = 'insert into talk_track (track_id, talk_id) values (:track_id, :talk_id)';
+            $stmt = $this->_db->prepare($sql);
+            $stmt->execute($params);
+        
+            $talk_track_id = $this->_db->lastInsertId();
+        }
+        return $talk_track_id;
+    }
+
+    /**
+     * Remove talk from a track
+     *
+     * @param  int $talk_id
+     * @param  int $track_id
+     */
+    public function removeTrackFromTalk($talk_id, $track_id)
```

Comment:
Either we're adding a talk to a track (then we also need to remove that talk from the track) or we add a track to a talk (then we also need to remove the track from the talk). Which way around?